packages/store-ui/src/molecules/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,197 @@
+
+  const transformValues = useMemo(
+    () => createTransformValues(infiniteMode, numberOfSlides),
+    [numberOfSlides, infiniteMode]
+  )
+
+  const { handlers, slide, sliderState, sliderDispatch } = useSlider({
+    totalItems: childrenCount,
+    itemsPerPage: 1,
+    infiniteMode,
+    slidingTransitionDuration: transition.duration,
+    ...swipeableConfigOverrides,
+  })
+
+  const { isItemVisible, shouldRenderItem } = useSlideVisibility({
+    itemsPerPage: sliderState.itemsPerPage,
+    currentSlide: sliderState.currentItem,
+    totalItems: childrenCount,
+  })
+
+  const postRenderedSlides =
+    infiniteMode && children ? childrenArray.slice(0, 1) : []
+
+  const preRenderedSlides =
+    infiniteMode && children ? childrenArray.slice(childrenCount - 1) : []
+
+  const slides = preRenderedSlides.concat(children ?? [], postRenderedSlides)
+
+  return (
+    <section data-store-carousel data-testid={testId} aria-label="carousel">
+      <div
+        data-carousel-track-container
+        style={{ overflow: 'hidden', width: '100%' }}
+        {...handlers}
+      >
+        <div
+          data-carousel-track
+          style={{
+            display: 'flex',
+            transition: sliderState.sliding ? slidingTransition : undefined,
+            width: `${numberOfSlides * 100}%`,
+            transform: `translate3d(${
+              transformValues[sliderState.currentPage]
+            }%, 0, 0)`,
+          }}
+          onTransitionEnd={() => {
+            if (sliderState.currentItem >= childrenCount) {
+              sliderDispatch({
+                type: 'GO_TO_PAGE',
+                payload: {
+                  pageIndex: 0,
+                  shouldSlide: false,
+                },
+              })
+            }
+
+            if (sliderState.currentItem < 0) {
+              sliderDispatch({
+                type: 'GO_TO_PAGE',
+                payload: {
+                  pageIndex: sliderState.totalPages - 1,
+                  shouldSlide: false,
+                },
+              })
+            }
+          }}
+        >
+          {slides.map((currentSlide, idx) => (
+            <div
+              key={idx}
+              data-carousel-item
+              style={{ width: '100%' }}
+              data-visible={
+                isItemVisible(idx - Number(infiniteMode)) || undefined
+              }
+            >
+              {shouldRenderItem(idx - Number(infiniteMode))
+                ? currentSlide
+                : null}
+            </div>
+          ))}
+        </div>
+      </div>
+
+      {showNavigationArrows && (
+        <div data-carousel-controls>
+          <Button
+            data-left-arrow
+            aria-controls="carousel"
+            aria-label="previous"
+            onClick={() => {
+              if (sliderState.sliding) {
+                return
+              }
+
+              slide('previous', transition.duration, sliderDispatch)
+            }}
```

Comment:
What do you think about extract this function to a new one that receives `previous` or `next`?

Reply:
You can go further and receive also a number, and use it on bullets, right?

Reply:
Yeeeah, that would work! And would remove the need for repetition, since in all of the `onClick` handlers I need to check for `sliderState.sliding`. But I don't know how I feel about creating another function that pretty much just calls `slide`. Even the function signature would be identical

Reply:
Ok ok, no problem.